apps/web/middleware.ts

```diff
@@ -32,6 +32,12 @@ const middleware: NextMiddleware = async (req) => {
     }
   }
 
+  // Allow visiting new booker page when cookie is present.
+  if (url.pathname.includes("/new-booker/") && !req.cookies.has("new-booker-enabled")) {
```

Comment:
Can we make invisible rewrite so people with the cookie can see the new booker without going to `"/new-booker/"`?

Reply:
@zomars  Great idea! Not sure how we would implement that though, because:

1. In middleware we would need to match ALL routes instead of only the paths in `config.matcher`, which would run a lot more middleware invocations? Plus not sure how we can detect if we're actually looking at a booking page because it only contains out of 2 dynamic parts? 
2. We also can't run it in geterversideprops, because `/[user]/[type]` is a statically generated page. 

Would love to hear how you would solve this.